config.yaml

```diff
@@ -3,28 +3,22 @@
 # See LICENSE file for licensing details.
 
 options:
-  zookeeper-properties:
-    description: zookeeper.properties configuration file.
+  additional-options:
+    description: "Newline-delimited list of additional custom options"
     type: string
-    default: |
-      clientPort=2181
-      dataDir=/var/snap/kafka/common/data
-      dataLogDir=/var/snap/kafka/common/log
-      dynamicConfigFile=/var/snap/kafka/common/zookeeper-dynamic.properties
-      tickTime=2000
-      initLimit=5
-      syncLimit=2
-      maxClientCnxns=60
-      minSessionTimeout=4000
-      maxSessionTimeout=40000
-      autopurge.snapRetainCount=3
-      autopurge.purgeInterval=0
-      reconfigEnabled=true
-      standaloneEnabled=false
-      4lw.commands.whitelist=*
-      DigestAuthenticationProvider.digestAlg=SHA3-256
-      quorum.auth.enableSasl=true
-      quorum.auth.learnerRequireSasl=true
-      quorum.auth.serverRequireSasl=true
-      authProvider.sasl=org.apache.zookeeper.server.auth.SASLAuthenticationProvider
-      audit.enable=true
+  client-port:
+    description: ""
+    type: int
+    default: 2181
+  data-dir:
+    description: ""
+    type: string
+    default: /var/snap/kafka/common/data
```

Comment:
If the snap is classically confined, then changing this is likely useless either way as the snap may not be able to access arbitrary paths on the system

Reply:
That's my bad, I had suggested it but forgotten this!

If a user wanted to use Juju storage, how would that work with confined snaps?

Reply:
I think we'd need to implement the [`system-files`](https://snapcraft.io/docs/system-files-interface) interface in the snap, pointing to the desired data directory on the host (e.g. `/var/lib/kafka`).

Then we can add a Juju storage in the `metadata.yaml` that is mounted at `/var/lib/kafka` (or wherever we choose)?

Reply:
OK, then I think it's probably best to remove these options for now, and we can work on adding `storage` support as a feature further down the line.

Reply:
Yeh, its something that'll need some careful thought (and probably a spec) :)